src/main/java/at/hannibal2/skyhanni/features/inventory/experimentationtable/SuperpairDataDisplay.kt

```diff
@@ -0,0 +1,257 @@
+                    anyDuplicates(it.key.pair.first.index, item.index, match.index) ||
+                        anyDuplicates(it.key.pair.first.index, item.index, match.index))
+            }) return
+
+        found.entries.removeIf { it.value == FoundType.NORMAL && anyDuplicates(it.key.item.index, item.index, match.index) }
+        found[FoundData(pair = pair)] = FoundType.MATCH
+    }
+
+    private fun handleNormalReward(item: Item) {
+        if (found.any {
+                when {
+                    it.value.isOneOf(FoundType.MATCH, FoundType.PAIR) ->
+                        anyDuplicates(item.index, it.key.pair.first.index, it.key.pair.second.index)
+
+                    else -> it.key.item.index == item.index && it.key.item.sameAs(item)
+                }
+            }) return
+
+        found[FoundData(item = item)] = FoundType.NORMAL
+    }
+
+    private fun drawDisplay() = buildList {
+        val currentExperiment = ExperimentationTableAPI.getCurrentExperiment() ?: return emptyList<String>()
+
+        add("§6Superpair Experimentation Data")
+        add("")
+
+        val normals = found.entries.filter { it.value == FoundType.NORMAL }
+        val powerups = found.entries.filter { it.value == FoundType.POWERUP }
+        val matches = found.entries.filter { it.value == FoundType.MATCH }
+        val pairs = found.entries.filter { it.value == FoundType.PAIR }
+        val possiblePairs = calculatePossiblePairs(currentExperiment)
+
+        if (pairs.isNotEmpty()) add("§2Found")
+        for (pair in pairs) {
+            val prefix = determinePrefix(pairs.indexOf(pair), pairs.lastIndex)
+            add(" $prefix §a${pair.key.pair.first.reward}")
+        }
+        if (matches.isNotEmpty()) add("§eMatched")
+        for (match in matches) {
+            val prefix = determinePrefix(matches.indexOf(match), matches.lastIndex)
+            add(" $prefix §e${match.key.pair.first.reward}")
+        }
+        if (powerups.isNotEmpty()) add("§bPowerUp")
+        for (powerup in powerups) {
+            val prefix = determinePrefix(powerups.indexOf(powerup), powerups.size - 1)
+            add(" $prefix §b${powerup.key.item.reward}")
+        }
+        val toAdd = mutableListOf<String>()
+        if (possiblePairs >= 1) toAdd.add("§ePairs - $possiblePairs")
+        if (2 - powerups.size >= 1) toAdd.add("§bPowerUps - ${2 - powerups.size}")
+        if (normals.isNotEmpty()) toAdd.add("§7Normals - ${normals.size}")
+
+        if (toAdd.isNotEmpty()) {
+            add("")
+            add("§4Not found")
+        }
+        for (string in toAdd) if (string != toAdd.last()) add(" ├ $string") else add(" └ $string")
+    }
+
+    private fun calculatePossiblePairs(currentExperiment: Experiment) =
+        ((currentExperiment.gridSize - 2) / 2) - found.filter { it.value != FoundType.POWERUP }.size
+
+    private fun convertToReward(item: ItemStack) = if (item.displayName.removeColor() == "Enchanted Book") item.getLore()[2].removeColor()
```

Comment:
please remove the whole function, instead work with neu internal names. there are already utils functions in item utils that can parse item lore + name to internal name, reuse them

Reply:
i can´t seem to find those utils

Reply:
have you tried `ItemStack.getInternalName()` yet? what does this function return here?

Reply:
would need to test, are you sure that Enchanting Exp has an internal name tho?

Reply:
Why does enchanting exp exist? :( This requires basically an abstract enchanting table loot item, with implementations to item stack item and enchanting item. Super overkill, though.

or you add support for enchanting xp in the item parse logic where we also scrap the item name for enchanting books